src/apply.jl

```diff
@@ -112,3 +112,34 @@ function apply!(table::T, t::Transform; cols=nothing, kwargs...)::T where T
 
     return table
 end
+
+"""
+    apply!!(A::AbstractArray, ::Transform; dims=1, kwargs...)
+
+Applies the [`Transform`](@ref) to each element of `A` and appends the result to `A`, if
+possible, creating a new array.
+"""
+#TODO: syntax would imply this is mutating. Should we make it so a new array isn't constructed?
```

Comment:
I am ok using `!!` but if we do we should try to follow the convention for what this already means in the wild, which I believe is mutating if possible, and if not it creates a new type, doing type promotions if needed

https://juliafolds.github.io/BangBang.jl/dev/

Reply:
If we did use `apply!!` to do that, what syntax would you suggest for and "apply and append" method?

Reply:
I dont think it excludes appending?

Reply:
I'm confused...the behaviour that BangBang follows doesn't do the "appending" step automatically AFAIK. It does type promotion instead (which is not exactly what we're looking for here)

Reply:
I'm partial to just adding a kwarg `append=true`

Reply:
Just to note there are more [use-cases](https://github.com/mcabbott/AxisKeys.jl/blob/master/src/push.jl#L76-L83) that I found in the wild where it suggests that `!!` is "appending" the input.

Reply:
It wouldn't be my preferred way of doing it but I can be convinced. Admittedly, it's because I'm biased against adding another keyword argument (c.f. `name`) and I quite like the elegance of `apply!!`.

Reply:
To me, including the append in `apply!!` is _bending_ the interface, because it's this extra thing that's not explicit in the BangBang doc, and I can't find a function in BangBang that's analogous. (Any relevant examples out side BangBang.jl ?). But it's not clearly breaking it either.

I also think there's reason to provide a definitely-not-mutating apply-and-append, so e.g. we have `apply_append` and also `apply_append!!`.

Reply:
Yeah I've cooled off the `apply!!` syntax. If you check slack you'll see that it comes with an implicit understanding of what it does so in this case we'd be deviating from that. I'm going with an `apply_append` method instead.